bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -738,6 +627,159 @@ public Image(Device device, ImageDataProvider imageDataProvider) {
 	init();
 }
 
+
+/**
+ * Upon zoom update, this method copies the image data and add disabled or grayed image depending on the flag
+ *
+ * @param device the device on which to create the image
+ * @param data the image data to create the image from (must not be null)
+ * @param flag the style, either <code>IMAGE_DISABLE</code> or <code>IMAGE_GRAY</code>
+ * @return new image data with valid styling properties as per the flag
+ */
+private ImageData copyImageDataForDisabledOrGrayImages(Device device, ImageData data, int flag) {
+	ImageData returnImageData = null;
+	switch (flag) {
+		case SWT.IMAGE_DISABLE: {
+			ImageData newData = applyDisableImageData(device, data, data.height, data.width);
+			returnImageData = newData;
+			break;
+		}
+		case SWT.IMAGE_GRAY: {
+			ImageData newData = applyGrayImageData(data, data.height, data.width);
+			returnImageData = newData;
+			break;
+		}
+		default: {
+			returnImageData = data;
+			break;
+		}
+	}
```

Comment:
Wouldn't this implementation be simpler and also sufficient?
```suggestion
	switch (flag) {
		case SWT.IMAGE_DISABLE: {
			return applyDisableImageData(device, data, data.height, data.width);
		}
		case SWT.IMAGE_GRAY: {
			return applyGrayImageData(data, data.height, data.width);
		}
		default: {
			return data;
		}
	}
```

Reply:
I was just confused because I found it unnecessary complex. But it's of course okay as it is and rather a matter of taste.